### PR TITLE
test(document-i18n): add comprehensive test suite

### PR DESCRIPTION
**Pull request by pedrobonamin**
## Summary

Adds comprehensive test coverage to `@sanity/document-internationalization` plugin in preparation for the upcoming migration from `{_key: <language>, value: <value>}` to `{_key: random, value: <value>, language: <language>}` structure.

- **13 test files** with **123 tests** total
- All tests passing
- Follows patterns established in PR #561 (internationalized-array tests)

## Test breakdown

| Category | Tests | Files |
|----------|-------|-------|
| Utility functions | 26 | createReference, excludePaths, separateReferences |
| Components | 45 | LanguageBadge, Warning, LanguagePatch, LanguageOption, LanguageManage |
| Actions | 33 | DeleteTranslationAction, DuplicateWithTranslationsAction, DeleteMetadataAction |
| Schema | 18 | translation metadata schema |
| Package exports | 1 | index.ts |

## Test infrastructure

- Vitest with jsdom environment
- React Testing Library for component tests
- `renderHook` for action tests (hooks-based DocumentActionComponent)
- Mock factories in `src/test/helpers.ts` for consistent test data
- `ThemeWrapper` and `createMockSanityClient` in `src/test/component-helpers.tsx`

## Related

- Linear: SAPP-2780
- Reference: PR #561

## Test plan

- [x] All 123 tests pass locally
- [x] Tests cover disabled states, edge cases, error handling
- [x] Schema tests verify preview.prepare output format

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/@sanity/document-internationalization/src/actions/DeleteTranslationAction.tsx [hunk 1]
```diff
@@ -31,7 +54,9 @@ export const DeleteTranslationAction: DocumentActionComponent = (props) => {
       operation = 'UNSET'
       translations.forEach((translation) => {
         tx.patch(translation._id, (patch) =>
-          patch.unset([`${TRANSLATIONS_ARRAY_NAME}[_key == "${documentLanguage}"]`]),
+          patch.unset([
+            `${TRANSLATIONS_ARRAY_NAME}[${LANGUAGE_FIELD_NAME} == "${documentLanguage}"]`,
```
Comment (pedrobonamin) [inline plugins/@sanity/document-internationalization/src/actions/DeleteTranslationAction.tsx]: this is preparing ourselves for the migration to `language` field

File: plugins/@sanity/document-internationalization/README.md [hunk 2]
```diff
@@ -199,6 +200,37 @@ defineField({
 }),
 ```
 
+### Document actions
```
Comment (RitaDias) [inline plugins/@sanity/document-internationalization/README.md]: A quick thought I had while reading this: do we need to see if we have any actual docs out there in the wild that we should add this to /update?
